### PR TITLE
feat(web/server): Add options to rerun job on all assets

### PR DESCRIPTION
**Pull request by alextran1502**
- Add dto to include option to rerun job on all asset
- Preserve name on exif extraction
- Remove unused schedule job

## Todo
- [ ] UI Change on job card

### CODE REVIEW

File: server/apps/immich/src/api-v1/job/job.service.ts [hunk 1]
```diff
@@ -115,4 +138,8 @@ export class JobService {
         throw new BadRequestException(`Invalid job id: ${jobId}`);
     }
   }
+
+  private getFileNameWithoutExtension(path: string): string {
+    return basename(path, extname(path));
+  }
```
Comment (jrasm91) [inline server/apps/immich/src/api-v1/job/job.service.ts:144]: We seem to have this or similar code scattered through the repo, we should move it to a util function instead.

File: server/immich-openapi-specs.json [hunk 2]
```diff
@@ -4610,10 +4610,15 @@
         "properties": {
           "command": {
             "$ref": "#/components/schemas/JobCommand"
+          },
+          "includeAllAssets": {
+            "type": "object",
```
Comment (jrasm91) [inline server/immich-openapi-specs.json:4615]: Should this be a boolean?

File: server/apps/immich/src/api-v1/job/dto/job-command.dto.ts [hunk 3]
```diff
@@ -9,4 +9,7 @@ export class JobCommandDto {
     enumName: 'JobCommand',
   })
   command!: string;
+
+  @IsOptional()
```
Comment (jrasm91) [inline server/apps/immich/src/api-v1/job/dto/job-command.dto.ts:13]: Can you add `@IsBoolean()`

**Review by jrasm91 — APPROVED** [review]
Looks good! I assume, the job card change will include an additional button/option to re-process all assets.

**Comment by alextran1502** [comment]
> Looks good! I assume, the job card change will include an additional button/option to re-process all assets.

Correct, playing around with redesigning the job card 😛

**Pull request merged**